internal/pkg/db/redis/queries.go

```diff
@@ -137,6 +139,10 @@ func getObjectsBySomeRange(conn redis.Conn, command string, key string, start in
 		}
 	}
 
+	if len(objects) == 1 && objects[0] == nil {
```

Comment:
This check was added because without it, the single nil element trips up https://github.com/edgexfoundry/edgex-go/pull/2094/files#diff-e0713026144dfb12cdabe915f0403939R96 and makes it error out.

Reply:
Does this affect the other `getObjectsBy...Range` functions? If so, can we fix them here or create issues so we do not forget to fix it?

Reply:
It would have to, they're just syntactic sugar for this function.

Truly, this feels like treating a side effect and not the underlying problem, but it seems like an underlying problem with the Redis driver code not returning an empty array when it doesn't find anything.

Reply:
The redigo driver is a very thin wrapper on top of raw Redis commands. Though according to the [RESP](https://redis.io/topics/protocol#array-reply)spec

> A client library API should return a null object and not an empty Array when Redis replies with a Null Array. This is necessary to distinguish between an empty list and a different condition (for instance the timeout condition of the BLPOP command).

This could be a bug in redigo. I've added a task to my backlog to dig into this.

Reply:
@andresrinivasan we have modified this part in another PR https://github.com/edgexfoundry/edgex-go/pull/2192/files
please help take a look.  thanks.

Reply:
Dug into this a bit more. TL;DR: The driver is right.

From the [MGET documentation](https://redis.io/commands/mget)

> Returns the values of all specified keys. For every key that does not hold a string value or does not exist, the special value nil is returned. Because of this, the operation never fails.

In action this looks like

```
127.0.0.1:6379> set foo bar
OK
127.0.0.1:6379> set baz mumble
OK
127.0.0.1:6379> mget x y z
1) (nil)
2) (nil)
3) (nil)
127.0.0.1:6379> mget foo baz zoo
1) "bar"
2) "mumble"
3) (nil)
```

If we're getting _nil_ results from `MGET`, we're seeing a referential integrity challenge. In other words, the list of ids retrieved from the sorted set don't all exist as keys. I don't think we'll be able to fix the referential integrity challenge soon so I recommend checking all the values returned by `MGET` before trying to use them. 

The correct example is from PR #2192 

```
	var result [][]byte
	if len(ids) > 0 {
		result, err = redis.ByteSlices(conn.Do("MGET", ids...))
		if err != nil {
			return nil, err
		}
	}

	for _, obj := range result {
		if obj != nil {
			objects = append(objects, obj)
		}
	}
```

Vs this incomplete solution as it doesn't check all the results from `MGET`

```
	if len(ids) > 0 {
		objects, err = redis.ByteSlices(conn.Do("MGET", ids...))
		if err != nil {
			return nil, err
		}
	}

	if len(objects) == 1 && objects[0] == nil {
		objects = [][]byte{}
	}
```